x-pack/plugins/remote_clusters/__jest__/client_integration/add/remote_clusters_add.test.ts

```diff
@@ -0,0 +1,248 @@
+            );
+          };
+
+          [...NON_ALPHA_NUMERIC_CHARS, ...ACCENTED_CHARS]
+            .filter(notInArray(['-', '_', ':']))
+            .forEach(expectInvalidChar);
+        });
+
+        test('should require a numeric "port" to be set', async () => {
+          await actions.saveButton.click();
+
+          actions.seedsInput.setValue('192.168.1.1');
+          expect(actions.getErrorMessages()).toContain('A port is required.');
+
+          actions.seedsInput.setValue('192.168.1.1:abc');
+          expect(actions.getErrorMessages()).toContain('A port is required.');
+        });
+      });
+
+      describe('proxy address', () => {
+        beforeEach(async () => {
+          await act(async () => {
+            ({ actions, component } = await setup());
+          });
+
+          component.update();
+
+          actions.connectionModeSwitch.toggle();
+        });
+
+        test('should only allow alpha-numeric characters and "-" (dash) in the proxy address "host" part', async () => {
+          await actions.saveButton.click(); // display form errors
+
+          const expectInvalidChar = (char: string) => {
+            actions.setProxyAddress(`192.16${char}:3000`);
+            expect(actions.getErrorMessages()).toContain(
+              'Address must use host:port format. Example: 127.0.0.1:9400, localhost:9400. Hosts can only consist of letters, numbers, and dashes.'
+            );
+          };
+
+          [...NON_ALPHA_NUMERIC_CHARS, ...ACCENTED_CHARS]
+            .filter(notInArray(['-', '_', ':']))
+            .forEach(expectInvalidChar);
+        });
+
+        test('should require a numeric "port" to be set', async () => {
+          await actions.saveButton.click();
+
+          actions.setProxyAddress('192.168.1.1');
+          expect(actions.getErrorMessages()).toContain('A port is required.');
+
+          actions.setProxyAddress('192.168.1.1:abc');
+          expect(actions.getErrorMessages()).toContain('A port is required.');
+        });
+      });
+    });
+  });
+  describe('on cloud', () => {
+    beforeEach(async () => {
+      await act(async () => {
+        ({ actions, component } = await setup(true));
+      });
+
+      component.update();
+    });
+
+    test('renders a switch for cloud and manual modes', () => {
+      expect(actions.cloudUrlSwitch.exists()).toBe(true);
+    });
+
+    test('renders no switch for sniff and proxy modes', () => {
+      expect(actions.connectionModeSwitch.exists()).toBe(false);
+    });
+    test('new cluster defaults to cloud url', () => {
+      expect(actions.cloudUrlSwitch.isChecked()).toBe(false);
+    });
+
+    describe('validation', () => {
+      test('cloud url is required', () => {
+        actions.saveButton.click();
+        expect(actions.getErrorMessages()).toContain('A url is required.');
+      });
+      test('server name is required', () => {
```

Comment:
Can this test be combined with the one below? I don't think the description as-is is 100% accurate, as it is only testing if the label appears, not that it is required.